### ISSUE TITLE
branch create error

### ISSUE THREAD

**Issue by snits** [stacked-git/stgit#317]
I get the following:

# stg br --create tpm-lenovo origin/master
info: Recording `origin/master` as parent branch
thread 'main' panicked at 'assertion failed: matches!(category, gix :: refs :: Category :: LocalBranch)', src/cmd/branch/create.rs:183:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

The branch is actually created though. I will update with more info this afternoon with version, the backtrace, and checking what happens when not in a worktree when I have some time.

**Comment by snits** [comment]
So this is with 2.2.3, and isn't specific to a worktree. I'm guessing based on the assertion failure it has to do with trying to create a branch based on a branch from a remote? I'll poke around and take a look.

```
# stg br --create testing origin/master
info: Recording `origin/master` as parent branch
thread 'main' panicked at 'assertion failed: matches!(category, gix :: refs :: Category :: LocalBranch)', src/cmd/branch/create.rs:183:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
# git br | grep testing
  testing
  testing-cleanup
# git br -D testing
Deleted branch testing (was 78b421b6a7c6).
# git br | grep testing
  testing-cleanup
# RUST_BACKTRACE=1 stg br --create testing origin/master
info: Recording `origin/master` as parent branch
thread 'main' panicked at 'assertion failed: matches!(category, gix :: refs :: Category :: LocalBranch)', src/cmd/branch/create.rs:183:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
   1: core::panicking::panic_fmt
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:114:5
   3: stg::cmd::branch::create::dispatch
   4: stg::cmd::branch::run
   5: stg::execute_command
   6: stg::execute_stgit_alias
   7: stg::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
# git br | grep testing                                 
  testing
  testing-cleanup
# stg br --delete testing
# cd ../linux
# stg br -l | grep testing                              
  s 	rhel9.3-iommu          | testing456
  s 	testing-cleanup        | clone of rhel8.9-iommu
# RUST_BACKTRACE=1 stg br --create testing origin/master
info: Recording `origin/master` as parent branch
thread 'main' panicked at 'assertion failed: matches!(category, gix :: refs :: Category :: LocalBranch)', src/cmd/branch/create.rs:183:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
   1: core::panicking::panic_fmt
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:114:5
   3: stg::cmd::branch::create::dispatch
   4: stg::cmd::branch::run
   5: stg::execute_command
   6: stg::execute_stgit_alias
   7: stg::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
# stg --version
Stacked Git 2.2.3
Copyright (C) 2005-2022 StGit authors
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
SPDX-License-Identifier: GPL-2.0-only
git version 2.38.1

# RUST_BACKTRACE=full stg br --create testing origin/master
info: Recording `origin/master` as parent branch
thread 'main' panicked at 'assertion failed: matches!(category, gix :: refs :: Category :: LocalBranch)', src/cmd/branch/create.rs:183:5
stack backtrace:
   0:     0x55cc3b72c78a - std::backtrace_rs::backtrace::libunwind::trace::ha9053a9a07ca49cb
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x55cc3b72c78a - std::backtrace_rs::backtrace::trace_unsynchronized::h9c2852a457ad564e
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55cc3b72c78a - std::sys_common::backtrace::_print_fmt::h457936fbfaa0070f
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x55cc3b72c78a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h5779d7bf7f70cb0c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x55cc3b7534fe - core::fmt::write::h5a4baaff1bcd3eb5
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/fmt/mod.rs:1232:17
   5:     0x55cc3b728565 - std::io::Write::write_fmt::h4bc1f301cb9e9cce
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/io/mod.rs:1684:15
   6:     0x55cc3b72c555 - std::sys_common::backtrace::_print::h5fcdc36060f177e8
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x55cc3b72c555 - std::sys_common::backtrace::print::h54ca9458b876c8bf
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x55cc3b72deaf - std::panicking::default_hook::{{closure}}::hbe471161c7664ed6
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:271:22
   9:     0x55cc3b72dbeb - std::panicking::default_hook::ha3500da57aa4ac4f
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:290:9
  10:     0x55cc3b72e458 - std::panicking::rust_panic_with_hook::h50c09d000dc561d2
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:692:13
  11:     0x55cc3b72e312 - std::panicking::begin_panic_handler::{{closure}}::h9e2b2176e00e0d9c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:581:13
  12:     0x55cc3b72cbf6 - std::sys_common::backtrace::__rust_end_short_backtrace::h5739b8e512c09d02
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/sys_common/backtrace.rs:150:18
  13:     0x55cc3b72e062 - rust_begin_unwind
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:579:5
  14:     0x55cc3b36d9c3 - core::panicking::panic_fmt::hf33a1475b4dc5c3e
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:64:14
  15:     0x55cc3b36da5d - core::panicking::panic::h9533b2fee90b999e
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/panicking.rs:114:5
  16:     0x55cc3b447c47 - stg::cmd::branch::create::dispatch::h027df5fa0f5b7e22
  17:     0x55cc3b5457a3 - stg::cmd::branch::run::h24e90d1669711b26
  18:     0x55cc3b3e361f - stg::execute_command::h5440acccdc0c0a82
  19:     0x55cc3b3e4304 - stg::execute_stgit_alias::h10426dc565081490
  20:     0x55cc3b3e2cce - stg::main::he7677beba63ef6c5
  21:     0x55cc3b4340e3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h696e5df4037c87d0
  22:     0x55cc3b3ac899 - std::rt::lang_start::{{closure}}::h535bc5b6c8ada158
  23:     0x55cc3b72080c - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hd6efcd3bec896f2c
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:287:13
  24:     0x55cc3b72080c - std::panicking::try::do_call::hce04e543bb1f4cbb
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:487:40
  25:     0x55cc3b72080c - std::panicking::try::h3342dd4e1f680968
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:451:19
  26:     0x55cc3b72080c - std::panic::catch_unwind::h148ce1e59ac0cee7
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panic.rs:140:14
  27:     0x55cc3b72080c - std::rt::lang_start_internal::{{closure}}::h25f9dda2057a67fe
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/rt.rs:148:48
  28:     0x55cc3b72080c - std::panicking::try::do_call::h7caaaeaf9401650b
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:487:40
  29:     0x55cc3b72080c - std::panicking::try::he7d15285746cbbc2
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panicking.rs:451:19
  30:     0x55cc3b72080c - std::panic::catch_unwind::h89fb4f50c0301fe0
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/panic.rs:140:14
  31:     0x55cc3b72080c - std::rt::lang_start_internal::h078acd489417d3c1
                               at /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/std/src/rt.rs:148:20
  32:     0x55cc3b3e5665 - main
  33:     0x7fe0f71ee510 - __libc_start_call_main
  34:     0x7fe0f71ee5c9 - __libc_start_main@GLIBC_2.2.5
  35:     0x55cc3b36e055 - _start
  36:                0x0 - <unknown>

```

**Comment by jpgrayson** [comment]
@snits, thanks for this issue report along with the follow-up details. I've pushed a fix in 7ab88b0. If you have the opportunity to try it out, I'd appreciate knowing how the fix fits with your workflow.

**Comment by snits** [comment]
Works for me
